src/actions/sources/select.js

```diff
@@ -128,7 +130,10 @@ export function selectLocation(
       source = getSourceFromId(getState(), location.sourceId);
     }
 
-    dispatch(addTab(source.url, 0));
+    await dispatch(addTab(source.url));
```

Comment:
- This is the line that's breaking mochitests.
- This also appears to be the line that's causing the perceived performance slowdown.